ui/selectors/transactions.js

```diff
@@ -146,7 +146,10 @@ export const transactionsSelector = createSelector(
   transactionSubSelector,
   selectedAddressTxListSelector,
   (subSelectorTxList = [], selectedAddressTxList = []) => {
-    const txsToRender = selectedAddressTxList.concat(subSelectorTxList);
+    const txsToRender = selectedAddressTxList
+      // Incoming transactions are already included in the subSelectorTxList
+      .filter((tx) => tx.type !== TransactionType.incoming)
```

Comment:
Also, is this is an issue with any `incoming` transaction or specifically those where the `from` and `to` are the same? Do we need to make the filter more specific?

Reply:
This is specifically those where the `from` and `to` are the same. But `selectedAddressTxList` is already picked by `from` property. So there are two valid option to check. Either `to` or type `incoming`. If it's easier to read I can also put all conditions